src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -56,12 +56,12 @@ private BoundExpression LowerDeclarationPattern(BoundDeclarationPattern pattern,
             if (pattern.IsVar)
             {
                 Debug.Assert(input.Type == pattern.LocalSymbol.Type);
-                var assignment = _factory.AssignmentExpression(_factory.Local(pattern.LocalSymbol), input);
+                var assignment = _factory.AssignmentExpression(pattern.VariableAccess, input);
```

Comment:
We should not be making such assumptions and we never do. Please lower the node.

---
In reply to: [77397389](https://github.com/dotnet/roslyn/pull/13535#discussion_r77397389) [](ancestors = 77397389,77386791)